<?xml version="1.0"?>
<robot name="quadrotor" xmlns:xacro="http://www.ros.org/wiki/xacro">

  <xacro:arg name="namespace" default="quad_0" />

  <link name="base_link">
    <pose>0 0 0.182466 0 0 0</pose>
    <inertial>
      <mass value="1.316"/>
      <inertia ixx="0.0128" ixy="0" ixz="0" iyy="0.0128" iyz="0" izz="0.0218" />
    </inertial>
    <collision name="collision">
      <geometry>
        <mesh filename="model://quadrotor/meshes/quadrotor_base.dae" />
      </geometry>
    </collision>
    <visual name="visual">
      <geometry>
        <mesh filename="model://quadrotor/meshes/quadrotor_base.dae" />
      </geometry>
    </visual>
  </link>

  <gazebo>
    <!-- 给无人机施加力和力矩 -->
    <plugin name="gazebo_ros_force" filename="libgazebo_ros_force.so">
      <ros>
        <namespace>/$(arg namespace)</namespace>
        <remapping>gazebo_ros_force:=gazebo_ros_force</remapping>
      </ros>
      <!-- Name of link within model which will receive the force -->
      <link_name>base_link</link_name>
      <!-- Frame where the force/torque will be applied (options: world; link)-->
      <force_frame>world</force_frame>
    </plugin>

    <!-- 提供无人机的位姿和速度信息 -->
    <plugin name="robot_pos" filename="libgazebo_ros_p3d.so">
      <ros>
        <namespace>/$(arg namespace)</namespace>
        <remapping>odom:=odom</remapping>
      </ros>
      <!-- Name of the link within this model whose pose will be published -->
      <body_name>base_link</body_name>
      <!-- Name of another link within this model to use as a reference frame.
           Remove the tag to use the world as a reference. -->
      <frame_name>world</frame_name>
      <!-- Update rate in Hz, defaults to 0.0, which means as fast as possible -->
      <update_rate>30</update_rate>
    </plugin>

  </gazebo>
</robot>
